thunderfish/sortfishtype.py

```diff
@@ -0,0 +1,189 @@
+    return suggestion, r_value
+
+
+def plot_type_detection(data, rate, peak_idx, trough_idx, peakdet_th, r_value_th, type_suggestion, r_vals):
+    """ Plots the best-window, a zoomed index of it and the r-values that determine whether fish is pulse or wave-type.
+
+    :param data: (array-like). amplitude data
+    :param rate: (float). sample-rate in Hz
+    :param peak_idx: (array of int). Array with peak indices
+    :param trough_idx: (array of int). Array with trough indices
+    :param peakdet_th: (float or array-like). threshold for peak detection
+    :param r_value_th: (float). If the median r-value is smaller than this threshold, the fish is pulse-; else wave-type
+    :param type_suggestion: (str). Fish-type suggested by the algorithm
+    :param r_vals: (array-like). Array with r-values (peak2trough / peak2peak)
+    """
+    import matplotlib.pyplot as plt
+    from matplotlib.patches import Rectangle
+
+    t = np.arange(len(data)) / rate
+    fs = 14
+
+    # Draw Figure with subplots
+    fig, ax = plt.subplots(nrows=3, ncols=1, figsize=(8., 12.))
+
+    # First and second subplots: best window with peak-trough-detection
+    for enu, c_axis in enumerate(ax):
+        if enu < 2:
+            c_axis.plot(t, data, lw=2.5, color='purple', alpha=0.7)
+            c_axis.plot(t[peak_idx], data[peak_idx], 'o', ms=11, mec='black', mew=1.5, color='crimson', alpha=0.7)
+            c_axis.plot(t[trough_idx], data[trough_idx], 'o', ms=11, mec='black', mew=1.5, color='lime', alpha=0.7)
+            c_axis.plot(t[::10], peakdet_th[::10], '-k', lw=2., rasterized=True, alpha=0.7)
+
+    # Define Inset Boundaries and plot them
+    inset_width = 20.  # in msec
+    start_in = int((len(data)/2. - (inset_width//2.*rate*0.001)))  # the 0.001 converts msec to sec
+    end_in = int((len(data)/2. + (inset_width//2.*rate*0.001)))
+    max_amp = np.max(data)
+    min_amp = np.min(data)
+
+    ax[0].add_patch(Rectangle((t[start_in], min_amp),  # (x,y)
+                              t[end_in]-t[start_in],  # width
+                              max_amp + abs(min_amp),  # height
+                              edgecolor='black', facecolor='white', lw=3))
+
+    # Cosmetics
+    ax[0].set_title('Best window with peak detection', fontsize=fs+2)
+    ax[0].set_xlabel('Time [sec]', fontsize=fs)
+    ax[0].set_ylabel('Amplitude [a.u.]', fontsize=fs)
+    ax[0].tick_params(axis='both', which='major', labelsize=fs - 2)
+
+    # Second Plot: Best-Window inset
+    ax[1].set_xlim(t[start_in], t[end_in])  # just set proper xlim!
+
+    # Cosmetics
+    ax[1].set_title('Best window Inset. Fish-type suggestion is %s' % type_suggestion, fontsize=fs + 2)
+    ax[1].set_xlabel('Time [sec]', fontsize=fs)
+    ax[1].set_ylabel('Amplitude [a.u.]', fontsize=fs)
+    ax[1].tick_params(axis='both', which='major', labelsize=fs - 2)
+
+    # Third Plot: r-values and Fish-type threshold
+    r_vals = r_vals[r_vals > 0]
```

Comment:
does this occur? If yes I would leave it in so that you see it in the histogram!